frontend/src/pages/modelServing/screens/projects/NIMServiceModal/NIMModelDeploymentNameSection.tsx

```diff
@@ -19,6 +20,10 @@ const NIMModelDeploymentNameSection: React.FC<NIMModelDeploymentNameSectionProps
       data-testid="model-deployment-name-section"
       value={data.name}
       onChange={(e, name) => setData('name', name)}
+      onBlur={() => {
+        const [translatedName] = translateDisplayNameForK8sAndReport(data.name);
+        setData('name', translatedName);
```

Comment:
Would this not just convert the field for what the user typed? Can we not support this under the hood and then supply it at the end? We have a typical two named set in our world... display name & k8s name -- we don't want to lose the display name to have a clean k8s name... let me see if I can see what guidance I can provide more specifically to solve this.

Reply:
@andrewballantyne This doesn't resolve the issue of allowing illegal characters. For example, if you enter the name "אקדא אעגכדע" (which contains no digits and includes a space), the problem persists.

Reply:
If you make a display name with those characters it should work fine 🤔 (or so I would think) -- k8s names cannot have it. Today we trim those out, which got you 12456 as a resulting username -- this had problems being just numbers.

What error are you getting? @olavtar

Reply:
This utility is used on the create project modal... this is the result of just using those Hebrew characters: 
<img width="356" alt="image" src="https://github.com/user-attachments/assets/88df39cf-5aae-4399-ae46-18e0c72a3ad1">

Reply:
Updated